### PR TITLE
New release workflow with support for ARM binaries

### LINKED ISSUE #411: build and release an arm64 version of hadolint
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

Previously #314 noting failures.

### Expected behavior

Releases directory and docker image published to include a linux/arm64 version of `hadolint`.

### Actual behavior

Only linux/amd64 versions are available.

### Steps to reproduce the behavior

See https://hub.docker.com/r/hadolint/hadolint/tags

**Additional environment details (OS, stack version, etc.)**

Please note that substantial progress has been made on Haskell for aarch64 (arm64), and the report I have from my colleague who has gone through this motivates this work by noting that `hadolint` is part of the build process for several Kubernetes components, and that it can be built by using Haskell's published `stack` binaries.

### PR DESCRIPTION
**Pull request by lorenzo**
The next release of hadolint will have ARM64 binaries with this change

closes #411

### CODE REVIEW

File: nix/pkgs.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    crossSystem = native.lib.systems.examples.musl64;
+  });
+
+  aarch64-static = import nixpkgsSrc (nixpkgsArgs // {
+    crossSystem = native.lib.systems.examples.aarch64-multiplatform-musl;
+  });
+
+  aarch64-darwin = import nixpkgsSrc (nixpkgsArgs // {
+    crossSystem = native.lib.systems.examples.aarch64-darwin;
+  });
```
Comment (m-ildefons) [inline nix/pkgs.nix:20]: I don't quite get why there is plumbing for cross-compiling for Darwin AArch64 here, but it's not used during release builds as far as I can tell.

**Review by m-ildefons — APPROVED** [review]
This is awesome!

File: nix/pkgs.nix (hunk 1 above)
Comment (lorenzo) [reply nix/pkgs.nix:20]: I'm working on that on my local machine, but wanted to release this now that I have the linux binaries working already

Comment (Fixmetal) [reply nix/pkgs.nix:20]: Darwin aarch64 Is about the new Apple M1 so it is definitely needed

Comment (lorenzo) [reply nix/pkgs.nix:20]: @Fixmetal you can already get it from homebrew

**Comment by potiuk** [comment]
Fantastic THANKS!

**Comment by potiuk** [comment]
Works LIKE A CHARM (including all the new `dockerfile:1.4` features! (And is LIGHTNIG FAST) on M1 :).

I just re-enabled it in Apache Airflow in https://github.com/apache/airflow/pull/22625 :D. Great job @lorenzo!

**Comment by Fixmetal** [comment]
Is there a plan for the new build release to be published? 😉

**Comment by Fixmetal** [comment]
> today, probably. @Fixmetal

♥️ Can't wait to see it running on my raspberry 4

**Comment by lorenzo** [comment]
@Fixmetal we are producing arm64 binaries, though. Not sure if raspberry is arm64?

**Comment by potiuk** [comment]
> @Fixmetal we are producing arm64 binaries, though. Not sure if raspberry is arm64?

Unless you use very old one (with arm32) - tt should be fine with Raspbery PI.

**Comment by Fixmetal** [comment]
> @Fixmetal we are producing arm64 binaries, though. Not sure if raspberry is arm64?

Yes, rasp4b has armv8 which is aarch64!

**Comment by potiuk** [comment]
Fun ting I just learned that the non-bet OS in RaspPi was mostly 32 till very recently: But 64 ARM architecture seems to be the only released since 2016 - and this is the imprtant bit :)

https://arstechnica.com/gadgets/2022/02/64-bit-raspberry-pi-os-exits-beta-is-available-for-all-pi-3-4-and-zero-2-boards/#:~:text=The%20Raspberry%20Pi%20hardware%20has,has%20remained%20primarily%2032%2Dbit.

**Pull request merged**